### PR TITLE
pass a TASK_BBOX to conflation when present

### PR DESCRIPTION
**Pull request by brianhatchl**
refs #1597

### CODE REVIEW

File: modules/Hoot/ui/sidebar/layerConflate.js [hunk 1]
```diff
@@ -212,6 +213,18 @@ class LayerConflate extends SidebarForm {
             delete data.HOOT2_ADV_OPTIONS.RoadEngines;
         }
 
+
+        //If a task grid is present in custom data, use it to restrict conflation
+        let customDataLayer = Hoot.context.layers().layer('data');
+        //Should we also check that a gpx url hash is present?
+        //Or should a manually added (not necessarily TM added)
+        //data layer be availabe for use?
```
Comment (brianhatchl) [inline modules/Hoot/ui/sidebar/layerConflate.js]: :point_up: @bobi7 what do you think?

File: modules/Hoot/ui/sidebar/layerConflate.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import SidebarForm                from './sidebarForm';
 import AdvancedOpts               from './advancedOpts';
 import FormFactory                from '../../tools/formFactory';
 import { layerConflateForm }      from '../../config/domMetadata';
+//import { utilQsString }           from '../../../util';
```
Comment (bobi7) [inline modules/Hoot/ui/sidebar/layerConflate.js]: delete

File: modules/Hoot/ui/sidebar/layerConflate.js (hunk 1 above)
Comment (bobi7) [reply modules/Hoot/ui/sidebar/layerConflate.js]: I like manually added data layer to be available for use

File: modules/Hoot/ui/sidebar/layerConflate.js [hunk 3]
```diff
@@ -212,6 +213,18 @@ class LayerConflate extends SidebarForm {
             delete data.HOOT2_ADV_OPTIONS.RoadEngines;
         }
 
+
+        //If a task grid is present in custom data, use it to restrict conflation
+        let customDataLayer = Hoot.context.layers().layer('data');
+        //Should we also check that a gpx url hash is present?
+        //Or should a manually added (not necessarily TM added)
+        //data layer be availabe for use?
+        //var q = utilStringQs(window.location.hash.substring(1));
```
Comment (bobi7) [inline modules/Hoot/ui/sidebar/layerConflate.js]: delete

File: modules/Hoot/ui/sidebar/layerConflate.js [hunk 4]
```diff
@@ -212,6 +213,18 @@ class LayerConflate extends SidebarForm {
...
+
+        if ( customDataLayer.hasData() && customDataLayer.enabled() /*&& q.gpx*/) {
```
Comment (bobi7) [inline modules/Hoot/ui/sidebar/layerConflate.js]: clean up comment

**Review by bobi7 — CHANGES_REQUESTED** [review]

**Review by bobi7 — APPROVED** [review]
💯

**Pull request merged**